src/pages/Compendia/DownloadCompendia.js

```diff
@@ -0,0 +1,185 @@
+}) => {
+  const agreedToTerms = !!token;
+  const [selected, setSelected] = React.useState(null);
+  const [agree, setAgree] = React.useState(agreedToTerms);
+  const { data, isLoading } = useLoader(() => fetchCompendiaData(filter));
+
+  if (isLoading)
+    return (
+      <div className="download-compendia">
+        <p className="download-compendia__title">Fetching {title}</p>
+        <Spinner />
+      </div>
+    );
+
+  if (!isLoading && data.length === 0)
+    return (
+      <div className="download-compendia">
+        Download unavailable at this time.
+        <br />
+        Please check again soon!
+      </div>
+    );
+
+  return (
+    <div className="download-compendia">
+      <div>
+        <div className="download-compendia__title">Download the {title}</div>
+        <div className="download-compendia__row">
+          <span className="download-compendia__choose">Choose Organism</span>
+          <Dropdown
+            multiple={false}
+            options={data}
+            selectedOption={selected}
+            label={c => formatSentenceCase(c.organism_name)}
+            onChange={s => setSelected(s)}
+          />
+        </div>
+
+        {children}
+
+        {!agreedToTerms && (
+          <div className="download-compendia__terms download-compendia__row">
+            <Checkbox
+              name="termsOfService"
+              checked={agree}
+              onClick={() => setAgree(!agree)}
+            >
+              I agree to the{' '}
+              <Link
+                to="/terms"
+                className="link"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                Terms of Use
+              </Link>
+            </Checkbox>
+          </div>
+        )}
+        <div className="download-compendia__row">
+          <span>
+            Download Size:{' '}
+            {getPrettyFileSize((selected || data[0]).size_in_bytes)}{' '}
+          </span>
+          <Button
+            text="Download Now"
+            className="download-compendia__button"
+            isDisabled={!agree}
+            onClick={async () => {
+              const tokenId = token || (await createToken());
+              const selectedOrganism = selected || data[0];
+              const downloadUrl = await downloadCompendia(
+                filter,
+                tokenId,
+                selectedOrganism
+              );
+              push({
+                pathname: '/compendia/download',
+                state: { organism: selectedOrganism, downloadUrl },
+              });
+            }}
+          />
+        </div>
+      </div>
+    </div>
+  );
+};
+DownloadCompendia = connect(
+  state => ({
+    token: state.token,
+  }),
+  { createToken, push }
+)(DownloadCompendia);
+export default DownloadCompendia;
+
+const filterForLatestCompendia = (data, organism = false) => {
+  data.forEach(c => {
+    c.organism_name = c.result.annotations[0].data.organism_name;
```

Comment:
Is there a way to achieve the same here without [mutating the parameter](https://github.com/airbnb/javascript#functions--mutate-params)?